### PR TITLE
fix: handle nil ParseableType from GVKParser

### PR DESCRIPTION
**Pull request by leoluz**
fix https://github.com/argoproj/argo-cd/issues/18020

### CODE REVIEW

File: pkg/diff/diff.go [hunk 1]
```diff
@@ -208,6 +208,10 @@ func removeWebhookMutation(predictedLive, live *unstructured.Unstructured, gvkPa
 	}
 	gvk := predictedLive.GetObjectKind().GroupVersionKind()
 	pt := gvkParser.Type(gvk)
+	if pt == nil {
+		return nil, fmt.Errorf("unable to resolve parseableType for gvk: %s", gvk)
```
Comment (todaywasawesome) [inline pkg/diff/diff.go:212]:
```suggestion
		return nil, fmt.Errorf("unable to resolve parseableType for gvk (Group Version Kind): %s", gvk)
```
I think we can make this a bit more clear.

File: pkg/diff/diff.go [hunk 2]
```diff
@@ -316,6 +320,9 @@ func structuredMergeDiff(p *SMDParams) (*DiffResult, error) {
 
 	gvk := p.config.GetObjectKind().GroupVersionKind()
 	pt := gescheme.ResolveParseableType(gvk, p.gvkParser)
+	if pt == nil {
+		return nil, fmt.Errorf("unable to resolve parseableType for gvk: %s", gvk)
```
Comment (todaywasawesome) [inline pkg/diff/diff.go:324]:
```suggestion
		return nil, fmt.Errorf("unable to resolve parseableType for gvk (Group Version Kind): %s", gvk)
```
Same

**Review by todaywasawesome — CHANGES_REQUESTED** [review]
If you're cool with it, let's update the text so it's a bit more clear for users.

**Review by crenshaw-dev — APPROVED** [review]

File: pkg/diff/diff.go (hunk 1 above)
Comment (leoluz) [reply pkg/diff/diff.go]: @todaywasawesome This is an internal error message but I agree with making it a bit more clear. I added a small modification. 
PTAL

File: pkg/diff/diff.go (hunk 2 above)
Comment (leoluz) [reply pkg/diff/diff.go]: @todaywasawesome This is an internal error message but I agree with making it a bit more clear. I added a small modification. 
PTAL

**Review by todaywasawesome — APPROVED** [review]

**Pull request merged**